File: backend/recpro-backend-lib/src/main/java/de/ubt/ai4/petter/recpro/recpro_backend_lib/recpro/model/recommendation/modeling/range/NumericRecommendationDimensionRange.java
package de.ubt.ai4.petter.recpro.recpro_backend_lib.recpro.model.recommendation.modeling.range;

import jakarta.persistence.DiscriminatorValue;
import jakarta.persistence.Entity;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.NoArgsConstructor;
import lombok.experimental.SuperBuilder;

@EqualsAndHashCode(callSuper = true)
@Entity
@Data
@SuperBuilder
@AllArgsConstructor
@NoArgsConstructor

@DiscriminatorValue("NUMERIC")
public class NumericRecommendationDimensionRange extends RecommendationDimensionRange {
    private double lowerBound;
    private double upperBound;
}

File: frontend/recpro-frontend-lib/projects/recpro-frontend-lib/src/lib/recommendation/modeling/range/model/NumericRecommendationDimensionRange.ts
import {RecommendationDimensionRange} from './RecommendationDimensionRange';
import {RecommendationDimensionRangeType} from './RecommendationDimensionRangeType';

export class NumericRecommendationDimensionRange extends RecommendationDimensionRange {
  lowerBound: number = 0;
  upperBound: number = 0;
  override rangeType = RecommendationDimensionRangeType.NUMERIC;
}
